### PR TITLE
XSUP-75731: Fix drilldown search parsing for unescaped backslashes

### PR DESCRIPTION
**Pull request by ilappe**
## Summary
- Fixed drilldown search parsing in SplunkPy and SplunkPy v2 where fetching incidents failed when placeholder values contained unescaped backslashes (e.g. `object="PREFIX\Admin"`), causing an `Invalid \escape` JSON parsing error
- Added unit tests covering the unescaped-backslash drilldown scenario for both integrations
- Bumped SplunkPy pack version and added ReleaseNotes 4.3.4

## Related Issues
fixes: XSUP-75731

### CODE REVIEW

**Comment by marketplace-ai-reviewer** [comment]
🤖 Analysis started. Please wait for results...

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy_test.py [hunk 1]
```diff
@@ -1963,6 +1964,95 @@ def test_escape_backslashes_in_field_filters(spl_search, expected):
...
+        # Already-doubled backslash (valid JSON) is left untouched (idempotent)
+        (
+            '{"search":"object=\\"a\\\\b\\""}',
+            '{"search":"object=\\"a\\\\b\\""}',
+        ),
+        # A unicode escape (\u) is a valid JSON escape and must be left untouched
+        (
+            '{"search":"snowman=\\u2603"}',
+            '{"search":"snowman=\\u2603"}',
+        ),
```
Comment (marketplace-ai-reviewer) [inline Packs/SplunkPy/Integrations/SplunkPy/SplunkPy_test.py:1994]: Consider adding test cases for backslashes followed by whitespace or at the very end of the string to ensure complete coverage of edge cases.

File: Packs/SplunkPy/Integrations/SplunkPyV2/SplunkPyV2_test.py [hunk 2]
```diff
@@ -1913,6 +1914,95 @@ def test_escape_backslashes_in_field_filters(spl_search, expected):
...
+        # Already-doubled backslash (valid JSON) is left untouched (idempotent)
+        (
+            '{"search":"object=\\"a\\\\b\\""}',
+            '{"search":"object=\\"a\\\\b\\""}',
+        ),
+        # A unicode escape (\u) is a valid JSON escape and must be left untouched
+        (
+            '{"search":"snowman=\\u2603"}',
+            '{"search":"snowman=\\u2603"}',
+        ),
```
Comment (marketplace-ai-reviewer) [inline Packs/SplunkPy/Integrations/SplunkPyV2/SplunkPyV2_test.py:1944]: Consider adding test cases for backslashes followed by whitespace or at the very end of the string to ensure complete coverage of edge cases.

File: Packs/SplunkPy/ReleaseNotes/4_3_4.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+
+#### Integrations
+
+##### Splunk
```
Comment (marketplace-ai-reviewer) [inline Packs/SplunkPy/ReleaseNotes/4_3_4.md]: Incorrect integration display name in the header.

File: Packs/SplunkPy/ReleaseNotes/4_3_4.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+
+- Fixed an issue where fetching incidents failed to parse drilldown search objects when placeholder values contained unescaped backslashes (for example, `object="PREFIX\Admin"`), causing an `Invalid \escape` JSON parsing error.
```
Comment (marketplace-ai-reviewer) [inline Packs/SplunkPy/ReleaseNotes/4_3_4.md]: Incorrect entity styling for literal values and error messages.

File: Packs/SplunkPy/ReleaseNotes/4_3_4.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+
+##### SplunkPy v2
+
+- Fixed an issue where fetching incidents failed to parse drilldown search objects when placeholder values contained unescaped backslashes (for example, `object="PREFIX\Admin"`), causing an `Invalid \escape` JSON parsing error.
```
Comment (marketplace-ai-reviewer) [inline Packs/SplunkPy/ReleaseNotes/4_3_4.md]: Incorrect entity styling for literal values and error messages.

**Comment by marketplace-ai-reviewer** [comment]
<details>
<summary><strong>🤖 AI Review Disclaimer</strong></summary>

This review was generated by an AI-powered tool and may contain inaccuracies. Please be advised, and we extend our sincere apologies for any inconvenience this may cause.
</details>

**Review by marketplace-ai-reviewer — COMMENTED** [review]
Hi there, thanks for your work on the Splunk integrations! I've reviewed the changes and just have a few minor suggestions to help polish things up. It would be great to add some edge-case test coverage for backslash handling in both versions of the integration, and there are a couple of quick formatting fixes needed in the release notes regarding the display name and text styling. Thanks again for the contribution!

 @kamalq97, @ilappe please review and approve the results generated by the AI Reviewer by responding 👍 on this comment.

**Review by kamalq97 — APPROVED** [review]

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy_test.py (hunk 1 above)
Comment (ilappe) [reply Packs/SplunkPy/Integrations/SplunkPy/SplunkPy_test.py:1994]: Done — added edge-case parametrized tests for a backslash followed by whitespace and a lone backslash near the end of a value.

File: Packs/SplunkPy/Integrations/SplunkPyV2/SplunkPyV2_test.py (hunk 2 above)
Comment (ilappe) [reply Packs/SplunkPy/Integrations/SplunkPyV2/SplunkPyV2_test.py:1944]: Done — added edge-case parametrized tests for a backslash followed by whitespace and a lone backslash near the end of a value.

File: Packs/SplunkPy/ReleaseNotes/4_3_4.md (hunk 3 above)
Comment (ilappe) [reply Packs/SplunkPy/ReleaseNotes/4_3_4.md]: Done — corrected the header to the integration display name (SplunkPy).

File: Packs/SplunkPy/ReleaseNotes/4_3_4.md (hunk 4 above)
Comment (ilappe) [reply Packs/SplunkPy/ReleaseNotes/4_3_4.md]: Done — reworded the release note to use a more general message without the flagged literal/error styling.

File: Packs/SplunkPy/ReleaseNotes/4_3_4.md (hunk 5 above)
Comment (ilappe) [reply Packs/SplunkPy/ReleaseNotes/4_3_4.md]: Done — reworded the release note to use a more general message without the flagged literal/error styling.
